src/http-api.js

```diff
@@ -679,6 +720,10 @@ module.exports.MatrixHttpApi.prototype = {
                     handlerFn(err, response, body);
                 }
             );
+            if (timeoutContext) {
+                req.onprogress = timeoutContext.onProgressCheck;
```

Comment:
How does this work as intended without `.bind()`ing, given you utilise `this` in `onProgressCheck`?